dbms/src/Flash/Mpp/MPPTunnelSetWriter.cpp

```diff
@@ -127,23 +128,41 @@ static void broadcastOrPassThroughWriteImpl(
     }
 
     // TODO avoid copy packet for broadcast.
-    for (size_t i = 0, local_cnt = 0, remote_cnt = 0; i < tunnel_cnt; ++i)
+
+    if (local_tracked_packet == remote_tracked_packet)
```

Comment:
how about `assert(local_tracked_packet != remote_tracked_packet)` and copy packet in
https://github.com/pingcap/tiflash/blob/0ff2b96178611511972333ad156f8fb4107be71b/dbms/src/Flash/Mpp/MPPTunnelSetWriter.cpp#L178

Reply:
Or
```
if (local_tracked_packet != nullptr && local_tracked_packet == remote_tracked_packet)
    local_tracked_packet = local_tracked_packet->copy();
```
It looks like the total number of copies is the same.

Reply:
Current way is aimed to reduce maximum memory usage.